tests/ui/self/arbitrary_self_types_pin_getref_default.rs

```diff
@@ -0,0 +1,22 @@
+// Confirms that Pin::get_ref can shadow methods in pointees, so long
+// as we don't enable arbitrary self types.
+// This will turn into an error if arbitrary_self_types feature is enabled.
+
+//@ run-pass
```

Comment:
This doesn't need to be run pass if it's not exercising any runtime behavior.

Also you should use revisions rather than making two tests.

Reply:
I couldn't figure out how to do that. This is what I know how to do:

```rust
//@ revisions: default feature
#![cfg_attr(feature, feature(arbitrary_self_types))]
```

but I want it to compile in one case, and not in the other. Even without the `//@ run-pass` it still complains at me unless _both_ revisions emit a compile error. Can you advise?

Reply:
it should be //@[feature] check-pass

and check-pass since as i said above you're not exercising runtime behavior

Reply:
@adetaylor please refer to <https://rustc-dev-guide.rust-lang.org/tests/compiletest.html#revisions>. You can do something like:

```rs
//@ revisions: disabled enabled
//@[disabled] check-pass

#![cfg_attr(enabled, arbitrary_self_types)]

fn whatever_line_should_error_when_feature_enabled() {
    //[enabled]~ ERROR meow meow meow
}
```

Note the `//@[revision_name]` revision-specific compiletest directive syntax, as well as the `//[revision_name]~` revision-specific error annotation syntax.

Reply:
Done, thanks